tools/cert_create/Makefile

```diff
@@ -61,8 +54,7 @@ all: clean ${BINARY}
 
 ${BINARY}: ${OBJECTS} Makefile
 	@echo "  LD      $@"
-	@echo 'const char build_msg[] = "Built : "__TIME__", "__DATE__; \
-                const char platform_msg[] = "${PLAT}";' | \
```

Comment:
Thanks for the extra info. I understand it's desirable to keep the host tools platform independent where possible. That should be achievable in the environments you care about but I don't think this is always going to be possible (e.g. for non-TBBR compliant platforms or platforms that really do want to use their own OIDs).

The device-side code currently has platform interfaces that enable selection of TBBR related code. Ideally the tools should have the same thing but this hasn't been implemented yet. Although the cert_create tool is currently hardcoded to use TBBR definitions, this may not be the case in future.

Also, I'm concerned about compatibility with platforms that do provide their own platform_oid.h, which would now be ignored by your patch.

I think a compromise solution is to allow the TBBR definitions to be used by default, while still allowing platforms to override if they want to. So something like the following:

- Moving the board_arm_oid.h definitions to include/tools_share/tbbr_oid.h (as you have done) looks OK.

- There needs to be a mechanism for a platforms's platform_oid.h to be used if required. There could be a build flag (e.g. USE_TBBR_DEFS), which controls whether platform_oid.h or tbbr_oid.h is used. This should default to false but could default to true if ERROR_DEPRECATED is also true.

- The upstream platforms can then enable USE_TBBR_DEFS and remove their platform_oid.h, although ideally this should come in a separate patch to prove this is backward compatible.

- cert_create should have the same mechanism. The build flag could also be used to control the platform_msg[] initalisation; when true the message could be hardcoded to "TBBR Generic" or similar.

- Potentially other parts of the codebase could use USE_TBBR_DEFS to control whether TBBR source files are pulled in, instead of leaving this to platforms to define. However, that is a bit off-topic and can come later.

Does that sound OK?